src/llvm-lower-handlers.cpp

```diff
@@ -197,14 +198,15 @@ static bool lowerExcHandlers(Function &F) {
             buff
         };
         CallInst::Create(lifetime_start, lifetime_args, "", new_enter);
-#ifndef _OS_WINDOWS_
-        // For LLVM 3.3 compatibility
-        Value *args[] = {buff,
-                         ConstantInt::get(Type::getInt32Ty(F.getContext()), 0)};
-        auto sj = CallInst::Create(setjmp_func, args, "", enter);
-#else
-        auto sj = CallInst::Create(setjmp_func, buff, "", enter);
-#endif
+        CallInst *sj;
+        if (!TT.isOSWindows()) {
+            // For LLVM 3.3 compatibility
+            Value *args[] = {buff,
+                            ConstantInt::get(Type::getInt32Ty(F.getContext()), 0)};
+            sj = CallInst::Create(setjmp_func, args, "", enter);
```

Comment:
```suggestion
            Value *args[] = {buff, ConstantInt::get(Type::getInt32Ty(F.getContext()), 0)};
            sj = CallInst::Create(setjmp_func, args, "", enter);
```

LLVM 3.3 did not have std::initializer_list, so we could not simply write this as, but that is also just a tediously long expression, so the current version is identical and equally reasonable.
```
            sj = CallInst::Create(setjmp_func, {buff, ConstantInt::get(Type::getInt32Ty(F.getContext()), 0)}, "", enter);
```

Or if we wanted to be exceptionally clever here (or with `.drop_back()` instead of the explicit length):
```
        Value *args[] = {buff, ConstantInt::get(Type::getInt32Ty(F.getContext()), 0)};
        sj = CallInst::Create(setjmp_func, ArrayRef(args, 2 - TT.isOSWindows()), "", enter);
```